### ISSUE TITLE
Inconsistent results

### ISSUE THREAD

**Issue by sam1am** [kenniy/fitbit-hooks#13]
It seems like you have to wait a while before your taps start working. Sometimes the vibration will come much later, so you will accidentally press the button multiple times. The feedback needs to be improved. Maybe don't show the buttons until a connection is confirmed? Or store the taps locally until they can be synchronized and confirmed?

**Comment by kenniy** [comment]
You are right. Basically the vibration is an acknowledgement for when a webhook has actually been sent. Sometimes, the companion app doesn't get connected immediately, so although you have tapped, the button, nothing gets sent.
It can probably be improved by greying out the button once pressed, but with the way Fitbit SDK works, it doesn't exactly get queued until you press again, and with some luck, the connection has been established. Guess a local implementation can be done to check first that the connection is established before the button is available for press, as you suggested. If you think that would make thing much better, can add that to next improvements.

**Comment by sam1am** [comment]
I think it would be an improvement. I am using the application as a counter (like for tracking how many glasses of water I have had or something), so if I tap the button once, wait a while with no response, tap it again, and then get two vibrations I've over-counted. Basically I can't trust it.   

My workaround in the meantime is to create a dummy button that doesn't really do anything. Tap on that until I start getting vibration results. Then tap the thing I'm actually going for. I imagine this could be done automatically, behind the scenes, as you suggest.
